packages/mdc-textfield/mdc-textfield.scss

```diff
@@ -313,6 +317,95 @@ $mdc-textfield-disabled-border-on-dark: rgba(white, .3);
   // stylelint-enable plugin/selector-bem-pattern
 }
 
+.mdc-textfield--with-leading-icon {
+  .mdc-textfield__leading-icon:not([tabindex]),
+  .mdc-textfield__leading-icon[tabindex="-1"] {
+    pointer-events: none;
+    cursor: default;
+  }
+
+  .mdc-textfield__trailing-icon:not([tabindex]),
+  .mdc-textfield__trailing-icon[tabindex="-1"] {
+    pointer-events: none;
+    cursor: default;
+  }
+
+  .mdc-textfield__input {
+    @include mdc-rtl-reflexive-box(padding, left, $mdc-textfield-icon-padding);
+  }
+
+  .mdc-textfield__leading-icon {
+    position: absolute;
+    bottom: 16px;
+    cursor: pointer;
+
+    @include mdc-rtl-reflexive-position(left, $mdc-textfield-icon-position);
+
+    @include mdc-theme-dark(".mdc-textfield") {
+      @include mdc-theme-prop(color, text-secondary-on-dark);
+    }
+  }
+
+  .mdc-textfield__label {
+    @include mdc-rtl-reflexive-position(left, $mdc-textfield-icon-padding);
+  }
+}
+
+.mdc-textfield--with-trailing-icon {
+  .mdc-textfield__trailing-icon {
+    position: absolute;
+    bottom: 16px;
+    width: 24px;
+    cursor: pointer;
+
+    @include mdc-rtl-reflexive-position(right, $mdc-textfield-icon-position);
+
+    @include mdc-theme-dark(".mdc-textfield") {
+      @include mdc-theme-prop(color, text-secondary-on-dark);
+    }
+  }
+
+  .mdc-textfield__label {
+    @include mdc-rtl(".mdc-textfield") {
+      padding-right: $mdc-textfield-icon-padding;
+    }
+  }
+}
+
+// stylelint-disable plugin/selector-bem-pattern
+.mdc-textfield--with-leading-icon.mdc-textfield--dense {
+  .mdc-textfield__input {
```

Comment:
This needs to be mirrored into the trailing dense styles; currently, if you enter a lot of text, it will flow right into the trailing icon.